### DISCUSSION TITLE
Facing issue when building app - Algorithm HmacPBESHA256 not available

### DISCUSSION THREAD

**Discussion by uphargaur** [oppia/oppia-android#5376] [Q&A (Installation)] [answered]
### Operating System

Linux

### Android Studio Version

Android Studio Bumblebee | 2021.1.1 Patch 3

### RAM size

16 gb

### Free disk space

400 gb

### Which step are you stuck on?

I  followed all the steps as mentioned in document , after the step of setting up Gradle Jdk version to 11 , i tried installing app in my physical device , there i got this issue when installing 

### Error Log

```bash
Execution failed for task ':app:packageDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.ide.common.signing.KeytoolException: Failed to read key AndroidDebugKey from store "/home/uphar/.android/debug.keystore": Integrity check failed: java.security.NoSuchAlgorithmException: Algorithm HmacPBESHA256 not available
```

### Approaches already used to resolve the issue

i tried to invalidate cache and clean project

**Comment by uphargaur** [comment]
issue is now resolved it was issue when u dont use proper jdk version

**Reply by adhiamboperes** [reply]
Glad to hear that you solved it, and apologies for missing this thread.

**Reply by uphargaur** [reply]
Thanks @adhiamboperes

**Reply by whyash8** [reply]
Please provide how you did it . I am having the same issue.

**Reply by adhiamboperes** [reply, marked as answer]
@whyash8, as @uphargaur mentioned, you need to use the correct jdk. In our case, that is jdk 11, and [here is a link](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android#prepare-developer-environment) to the wiki section explaining how to configure it. Please let us know if we can assist further.

**Reply by uphargaur** [reply]
@adhiamboperes  yes

**Reply by whyash8** [reply]
Thanks a lot , it's solved.
